### PR TITLE
Implemented NSProcessInfo's processName and tests.

### PR DESCRIPTION
**Pull request by DonaldHays**
This pull request implements the `processName` property on `NSProcessInfo`, and adds tests for that property.

Since `NSProcessInfo` didn’t have any tests already, I created the test class myself. Of note, pull request #50 also creates the same test class.

The unit test checks the initial value of `processName`, which is “TestFoundation”. That check may be inappropriate, since renaming the test project will break that test.

### CODE REVIEW

File: Foundation.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -539,7 +540,6 @@
 		848A30571C137B3500C83206 /* TestNSHTTPCookie.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; name = TestNSHTTPCookie.swift; path = TestFoundation/TestNSHTTPCookie.swift; sourceTree = SOURCE_ROOT; };
 		84BA558D1C16F90900F48C54 /* TestNSTimeZone.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = TestNSTimeZone.swift; sourceTree = "<group>"; };
 		88D28DE61C13AE9000494606 /* TestNSGeometry.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = TestNSGeometry.swift; sourceTree = "<group>"; };
-		C2A9D75B1C15C08B00993803 /* TestNSUUID.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = TestNSUUID.swift; sourceTree = "<group>"; };
```
Comment (jpsim) [inline Foundation.xcodeproj/project.pbxproj]: Did you accidentally remove this?

Comment (DonaldHays) [inline Foundation.xcodeproj/project.pbxproj]: The exact same line still appears again a few lines down. I think it was just duplicated and Xcode cleaned up the duplicates automatically.

File: TestFoundation/TestNSProcessInfo.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    var allTests : [(String, () -> ())] {
+        return [
+            ("test_processName", test_processName ),
+        ]
+    }
+    
+    func test_processName() {
+        // Assert that the original process name is "TestFoundation". This test
+        // will fail if the test target ever gets renamed, so maybe it should
+        // just test that the initial name is not empty or something?
```
Comment (phausler) [inline TestFoundation/TestNSProcessInfo.swift]: eh probably a safe assumption that it will still be named that

**Comment by DonaldHays** [comment]
Messed up the rebase, will close and re-open.

**Pull request closed without merging**